src/WCSimGenerator_Radioactivity.cc

```diff
@@ -271,1170 +142,525 @@ void WCSimGenerator_Radioactivity::Configuration(G4int iScenario, G4double dLife
+	double dConcentration_Above = dConc0 + (dConc10b - dConc0) * DiffusionZ(lZ,dLimit_PeakRight,dFactorR) + dConcentration_PMT;
 	
-	return  dConc;
-}
-
-G4double WCSimGenerator_Radioactivity::R2Fit_Z14p(G4double x, G4double Lambda, G4double BinConversion) {
+	//std::cout << "[DEBUG] RadonFormulaZ z= " << lZ << " " << dConcentration_Peak << " " << dConcentration_Below << " " << dConcentration_Above << std::endl;
 	
-	G4double dConc0      = pow(BinConversion,2.) * fRnSK_Center;	//   0m
-//	G4double dConc10     = pow(BinConversion,2.) * fRnSK_Bottom;	// Border (PMT)
-	G4double dConcPMT    = pow(BinConversion,2.) * fRn_Border;	// Border (PMT)
-		
-	G4double dConc       = dConc0*2; 								  // Center
-			   //+ dConcPMT        / cosh( sqrt(Lambda/(fRnDiffusion_Coef*4.))   * (sqrt(fChangeRMax) - sqrt(x)) )   // PMT
-			   //+ dConcPMT * 0.05 / cosh( sqrt(Lambda/(fRnDiffusion_Coef*250.))  * (sqrt(fChangeRMax) - sqrt(x)) );   // PMT
-	
-	// Water flow assumed		   
-	/*
-	double dConcWall   = dConc10 * 40;
-	if ( x >= fChangeZp14Flow ) {
-		dConc      += dConcWall / cosh( sqrt(Lambda/(fRnDiffusion_Coef*5.)) * (sqrt(x) - sqrt(fChangeZp14Flow)) ); 
+	if ( lZ < dLimit_PeakLeft ) {
+		return dConcentration_Below;  
 	}
-	else {
-		dConc      += dConcWall / cosh( sqrt(Lambda/(fRnDiffusion_Coef*5.)) * (sqrt(fChangeZp14Flow) - sqrt(x)) ); 
-	}
-	*/
-	// Water flow with convection is assumed
-	if ( x >= fChangeZp14Flow ) {
-		dConc      += dConcPMT * 1.22;
-	}
-	else {
-		dConc	   += dConcPMT * 1.2       / cosh( sqrt(Lambda/(fRnDiffusion_Coef*5)) * (sqrt(fChangeZp14Flow) - sqrt(x)) )
-			   + dConcPMT * 0.02 / cosh( sqrt(Lambda/(fRnDiffusion_Coef*200.))  * (sqrt(fChangeZp14Flow) - sqrt(x)) );   // PMT
+	else if ( lZ < dLimit_PeakRight ) {
+		return dConcentration_Peak;
 	}
 	
-	return  dConc;
+	return dConcentration_Above;
 }
 
 
-G4double WCSimGenerator_Radioactivity::R2Fit_Z16p(G4double x, G4double Lambda, G4double BinConversion) {
+//*****************************************************************************************************
+double WCSimGenerator_Radioactivity::RadonFormula(double *val, double *par) {
 
+	double lR2 	= val[0];
+	double lZ 	= val[1];
+	//double lR	= sqrt(lR2);
 	
-	G4double dConc0      = pow(BinConversion,2.) * fRnSK_Center;		//   0m
-	G4double dConc10     = pow(BinConversion,2.) * fRn_Border * 0.06;	// -10m (12m)
-	G4double dConcPMT    = pow(BinConversion,2.) * fRn_Border;		// Border (PMT)
-		
-	G4double dConc       = dConc0; 								  // Center
-				//+ dConcPMT        / cosh( sqrt(Lambda/(fRnDiffusion_Coef*4.)) * (sqrt(fChangeRMax) - sqrt(x)) )   // PMT
-				//+ dConcPMT * 0.05 / cosh( sqrt(Lambda/(fRnDiffusion_Coef*150.)) * (sqrt(fChangeRMax) - sqrt(x)) );   // Diffusion
-	
-	double dNorm10 = max((dConc10-dConc0),0.);
+	//----------------------------------//
+	// Make value arrays:
+	double vZ [2]	= { lZ , lR2	};
+	double vR2[2]	= { lR2, lZ 	};
+	//----------------------------------//
+	fRnLambda_Global = par[0];
```

Comment:
Again, I might be missing something, but where is `par[1]` used?
From the header file (and top of this file)
```par[1] is the M / bin```

Reply:
Ah, indeed, it's an artifact of the old version, I removed it I think, I will change the comments.